source/grid/grid_in.cc

```diff
@@ -2760,10 +2787,519 @@ GridIn<dim, spacedim>::parse_format (const std::string &format_name)
 template <int dim, int spacedim>
 std::string GridIn<dim, spacedim>::get_format_names ()
 {
-  return "dbmesh|msh|unv|vtk|ucd|xda|netcdf|tecplot";
+  return "dbmesh|msh|unv|vtk|ucd|abaqus|xda|netcdf|tecplot";
 }
 
 
+namespace internal
+{
+  template <int dim>
+  Abaqus_to_UCD<dim>::Abaqus_to_UCD (void)
```

Comment:
That's what we've typically done -- create a sub-namespace `internal::GridIn` that matches the name of the class where the internal stuff is used.

In the current case, are you actually using this class anywhere in the `.h` file? If not, you could also just move it to the `.cc` file and pack it into an anonymous (unnamed) namespace.